lib/cylc/gui/app_gcylc.py

```diff
@@ -613,11 +582,9 @@ def __init__(self, suite, db, owner, host, port, pyro_timeout,
         bigbox.pack_start(self.views_parent, True)
 
         if self.restricted_display:
-            self.legal_task_states = TaskState.legal_for_restricted_monitoring
-        else:
-            self.legal_task_states = TaskState.legal
+            self.legal_task_states = TASK_STATUSES_RESTRICTED
```

Comment:
I think `self.legal_task_states` needs to be set regardless. I am getting a traceback on `View > Task Filtering`.

```
Traceback (most recent call last):
  File "/home/matt/cylc.git/lib/cylc/gui/app_gcylc.py", line 3150, in popup_filter_dialog
    self.create_task_filter_widgets()
  File "/home/matt/cylc.git/lib/cylc/gui/app_gcylc.py", line 2911, in create_task_filter_widgets
    n_states = len(self.legal_task_states)
AttributeError: 'ControlApp' object has no attribute 'legal_task_states'

```